File: frontend/src/types/site.ts
export enum DbMode {
  Shared = 1,
  Dedicated = 2,
}

export interface SiteSummaryDto {
  id: string
  name: string
  city?: string
  district?: string
  isActive: boolean
  dbMode: DbMode
  blockCount: number
  unitCount: number
  createdAt: string
}

export interface SiteDetailDto {
  id: string
  name: string
  address?: string
  district?: string
  city?: string
  postalCode?: string
  phone?: string
  email?: string
  taxOffice?: string
  taxNumber?: string
  dbMode: DbMode
  isActive: boolean
  blockCount: number
  unitCount: number
  createdAt: string
  updatedAt: string
}

export interface CreateSiteDto {
  name: string
  dbMode: DbMode
  adminFirstName: string
  adminLastName: string
  adminEmail: string
  adminPhone?: string
  address?: string
  district?: string
  city?: string
  postalCode?: string
  phone?: string
  email?: string
  taxOffice?: string
  taxNumber?: string
}

export interface UpdateSiteDto {
  name: string
  address?: string
  district?: string
  city?: string
  postalCode?: string
  phone?: string
  email?: string
  taxOffice?: string
  taxNumber?: string
  isActive: boolean
}

File: backend/src/SiteYonetimi.Infrastructure/Entities/Site.cs
using SiteYonetimi.Shared.Entities;
using SiteYonetimi.Shared.Enums;

namespace SiteYonetimi.Infrastructure.Entities;

public class Site : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Address { get; set; }
    public string? District { get; set; }
    public string? City { get; set; }
    public string? PostalCode { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? TaxOffice { get; set; }
    public string? TaxNumber { get; set; }
    public DbMode DbMode { get; set; } = DbMode.Shared;
    public string? ConnectionString { get; set; } // Dedicated ise dolu, Shared ise null
    public bool IsActive { get; set; } = true;

    // Navigation
    public ICollection<UserSite> UserSites { get; set; } = new List<UserSite>();
    public ICollection<RoleType> RoleTypes { get; set; } = new List<RoleType>();
    public ICollection<SiteSubscription> SiteSubscriptions { get; set; } = new List<SiteSubscription>();
}
